src/alexapi/device_platforms/hyperion.py

```diff
@@ -0,0 +1,160 @@
+
+		self.host = self._pconfig['hyperion_json_host']
+		self.port = self._pconfig['hyperion_json_port']
+		self.service = ''
+		self.setup_complete = False
+		self.socket = None
+		self.socket_thread = None
+
+	def setup(self):
+		self.service = "ws://%s:%s" % (self.host, self.port)
+		logger.debug("Hyperion JSON Server - %s", self.service)
+		self.init_connection()
+
+	def after_setup(self, trigger_callback=None): # pylint: disable=unused-argument
+		self.setup_complete = True
+		self.check_connection()
+
+	def check_connection(self):
+		logger.info('Checking Hyperion Connection')
+		if self.socket_status():
+			logger.info("Connection OK")
+		else:
+			logger.error("Connection Failed")
+
+	def display_state(self, state):
+		return 'start' if state else 'stop'
+
+	def force_recording(self):
+		pass
+
+	def get_color(self, mode):
+		return self._pconfig["color_{}".format(mode)]
+
+	def handle_indicate(self, mode, state=False, flash=False):
+		logger.debug("indicate %s %s", mode, self.display_state(state))
+		flash = self.should_flash(mode)
+		if not state and not flash:
+			self.hyperion_clear()
+		if state:
+			self.hyperion_indicate(self.get_color(mode), flash)
+
+	def hyperion_clear(self):
+		logger.debug("Clearing Hyperion settings")
+		self.hyperion_send(self.hyperion_message('clear', True))
+
+	def hyperion_effect(self, color, flash=False):
+		effect = {'args': {'color': color}}
+		if flash:
+			effect['name'] = "Strobe white"
+			effect['args']['frequency'] = self._pconfig['flash_frequency']
+		else:
+			effect['name'] = "Knight rider"
+			effect['args']['speed'] = self._pconfig['hyperion_effect_speed']
+		return {'effect': effect}
+
+	def hyperion_indicate(self, color, flash=False, duration=False):
+		command = self._pconfig['hyperion_mode']
+		if flash and command == 'color':
+			command = 'effect'
+		if flash:
+			duration = self._pconfig['flash_duration']
+		options = self.hyperion_options(command, color, duration, flash)
+		if self.setup_complete:
+			self.hyperion_send(self.hyperion_message(command, True, options))
+
+	def hyperion_options(self, command, color, duration=False, flash=False):
+		options = {'color': color}
+		if command == 'effect':
+			options = self.hyperion_effect(color, flash)
+		if duration:
+			options['duration'] = duration
+		return options
+
+	def hyperion_message(self, command, priority=False, options=None):
+		message = options or {}
+		message['command'] = command
+		if priority:
+			message['priority'] = self._pconfig['hyperion_priority']
+		return message
+
+	def hyperion_send(self, message):
+		if self._pconfig['verbose']:
+			logger.debug("sending '%s'", json.dumps(message))
+		if self.socket_status():
+			self.socket.send(json.dumps(message))
+		else:
+			logger.warning("Unable to send Hyperion state update")
+			self.init_connection()
+
+	def indicate_failure(self):
+		pass
+
+	def indicate_playback(self, state=True):
+		if self._pconfig['indicate_playback']:
```

Comment:
Was just a decision by me because you either have
the sound response for that state or just pressed the trigger button. So
for me it was important that the recording and processing state always have
some indication. But as said, was just a personal decision and can be
changed to allow more customization.